File: missing-values-v2/code/python/regression.py
import numpy as np
from sklearn import  linear_model



# Load
training_file = '../../dataset/diabetes/training/training.csv'
test_file = '../../dataset/diabetes/test/test.csv'
training = np.genfromtxt(training_file, delimiter = ',')
test = np.genfromtxt(test_file, delimiter = ',')

X = training[:,:-1]
Y = training[:,-1]
# Linear regression
regr = linear_model.LinearRegression()
regr.fit(X, Y)

print('---------------------------------------------------')
print('Coefficients:')
print(regr.coef_)

# Test
X_test = test[:,:-1]
Y_test = test[:,-1]
print('---------------------------------------------------')
print('Mean squared error:')
print(np.mean((regr.predict(X_test) - Y_test) ** 2))
print('---------------------------------------------------')
print('Variance score:')
print(regr.score(X_test, Y_test))

File: missing-values-v2/code/regression.m
trainingPath = '../dataset/diabetes/training/training.csv';
testPath = '../dataset/diabetes/test/test.csv';

training = dlmread(trainingPath);
test = dlmread(testPath);

X = training(:, 1:10); 
Y = training(:, 11); 

Xtest = test(:, 1:10);
Ytest = test(:, 11);

d = size(X,2); % Number of features.

n = length(Y);    % Number of training examples
m = length(Ytest);% Number of test examples

Xo = ones(n,1);
XoTest = ones(m,1);

X = [Xo X];
Xtest = [XoTest Xtest];

theta = (pinv(X' * X) * X' * Y)'

Ypredict = sum((Xtest.*theta)')';
MSE = mean((Ypredict-Ytest).^2)
MSRE = sqrt(MSE)
